File: 排序/选择排序/select_sort.js
function select_sort (numbers) {
    let size = numbers.length;
    for (let i = 0; i < size - 1; i++) {
        let smallest_index = i;
        for (let j = smallest_index + 1; j < size; j++) {
            if (numbers[j] < numbers[smallest_index]) {
                smallest_index = j;
            }
        }
        if (smallest_index !== i) {
            [numbers[smallest_index], numbers[i]] = [numbers[i], numbers[smallest_index]];
        }
    }
}

File: 排序/选择排序/select_sort.py
buf = [1, 10, 20, 4, 8]
for i in range(len(buf) - 1):
    smallest_index = i
    for j in range(i + 1, len(buf)):
        if buf[j] < buf[smallest_index]:
            smallest_index = j
    if smallest_index != i:
        buf[i], buf[smallest_index] = buf[smallest_index], buf[i]
print(buf)